File: src/main/java/org/intellij/sdk/language/jass/JassRefactoringSupportProvider.java
package org.intellij.sdk.language.jass;

import com.intellij.lang.refactoring.RefactoringSupportProvider;
import com.intellij.psi.PsiElement;
import org.intellij.sdk.language.jass.psi.JassProperty;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

final class JassRefactoringSupportProvider extends RefactoringSupportProvider {

    @Override
    public boolean isMemberInplaceRenameAvailable(@NotNull PsiElement elementToRename, @Nullable PsiElement context) {
        return (elementToRename instanceof JassProperty);
    }
}

File: src/main/kotlin/raft/war/language/jass/JassRefactoringSupportProvider.kt
package raft.war.language.jass

import com.intellij.lang.refactoring.RefactoringSupportProvider
import com.intellij.psi.PsiElement
import raft.war.language.jass.psi.JassNamedElement

internal class JassRefactoringSupportProvider : RefactoringSupportProvider() {
    override fun isMemberInplaceRenameAvailable(elementToRename: PsiElement, context: PsiElement?): Boolean {
        return (elementToRename is JassNamedElement)
    }
}
